nuxeo-core/nuxeo-core-api/src/test/java/org/nuxeo/ecm/core/api/impl/TestDocumentModelWithSecuredProperty.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.api.impl;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.local.DummyLoginFeature;
+import org.nuxeo.ecm.core.api.local.WithUser;
+import org.nuxeo.ecm.core.api.model.ReadOnlyPropertyException;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+/**
+ * @since 9.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ RuntimeFeature.class, DummyLoginFeature.class })
+@Deploy("org.nuxeo.ecm.core.schema")
+@Deploy("org.nuxeo.ecm.core.api.tests:OSGI-INF/test-documentmodel-secured-types-contrib.xml")
+@WithUser("john")
+public class TestDocumentModelWithSecuredProperty {
+
+    protected static final String SECURED_SCHEMA = "secured";
+
+    @Test
+    public void testSetSecuredScalarProperty() {
+        try {
+            DocumentModel doc = new DocumentModelImpl("/", "doc", "Secured");
+            doc.setProperty(SECURED_SCHEMA, "scalar", "test secure");
+            fail("A ReadOnlyPropertyException should have been thrown");
+        } catch (ReadOnlyPropertyException e) {
+            assertEquals("scalar", e.getMessage());
```

Comment:
Not directly related to this PR, but I wonder if the message couldn't be more precise than just the property's path, maybe something like "Cannot set the value of property XPATH since it is readonly"?